readabilitySAX.js

```diff
@@ -491,18 +494,18 @@ Readability.prototype.onclosetag = function(tagName){
 	else if(this._settings.cleanConditionally && tagName in cleanConditionally){
 		var p = elem.info.tagCount.p || 0,
 		    contentLength = elem.info.textLength + elem.info.linkLength;
-
+        
 		if(contentLength === 0){
 			if(elem.children.length === 0) return;
 			if(elem.children.length === 1 && typeof elem.children[0] === "string") return;
 		}
 		if((elem.info.tagCount.li - 100) > p && tagName !== "ul" && tagName !== "ol") return;
-		if(contentLength < 25 && (!("img" in elem.info.tagCount) || elem.info.tagCount.img > 2) ) return;
+		if(contentLength < 25 && ((!("img" in elem.info.tagCount) && !("iframe" in elem.info.tagCount)) || elem.info.tagCount.img > 2 || elem.info.tagCount.iframe > 2) ) return;
```

Comment:
Not sure I like this, the image part is often annoying on its own.

It might be interesting to see how Safari Reader is handling this. I wanted to have a look at their algorithm changes for quite some time now, this might be an interesting starting point.